File: src/types/habit.ts
export interface Habit {
  id: string;
  name: string;
  color: string;
  icon: string;
  frequency: string;
  target_count: number;
  sort_order: number;
  created_at: string;
  updated_at: string;
}

export interface HabitWithStats extends Habit {
  current_streak: number;
  best_streak: number;
  completion_rate: number;
  is_done_today: boolean;
}

export interface HabitLog {
  id: string;
  habit_id: string;
  log_date: string;
  count: number;
  note: string;
  created_at: string;
}

export interface CreateHabitInput {
  name: string;
  color?: string;
  icon?: string;
  frequency?: string;
  target_count?: number;
}

export interface UpdateHabitInput {
  name?: string;
  color?: string;
  icon?: string;
  frequency?: string;
  target_count?: number;
}

export interface ReorderHabitsItem {
  id: string;
  sort_order: number;
}

File: src-tauri/src/models/habit.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Habit {
    pub id: String,
    pub name: String,
    pub color: String,
    pub icon: String,
    pub frequency: String,
    pub target_count: i32,
    pub sort_order: i32,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct HabitWithStats {
    pub id: String,
    pub name: String,
    pub color: String,
    pub icon: String,
    pub frequency: String,
    pub target_count: i32,
    pub sort_order: i32,
    pub created_at: String,
    pub updated_at: String,
    pub current_streak: i32,
    pub best_streak: i32,
    pub completion_rate: f64,
    pub is_done_today: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct HabitLog {
    pub id: String,
    pub habit_id: String,
    pub log_date: String,
    pub count: i32,
    pub note: String,
    pub created_at: String,
}

#[derive(Debug, Deserialize)]
pub struct CreateHabitRequest {
    pub name: String,
    pub color: Option<String>,
    pub icon: Option<String>,
    pub frequency: Option<String>,
    pub target_count: Option<i32>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateHabitRequest {
    pub name: Option<String>,
    pub color: Option<String>,
    pub icon: Option<String>,
    pub frequency: Option<String>,
    pub target_count: Option<i32>,
}

#[derive(Debug, Deserialize)]
pub struct ReorderHabitsItem {
    pub id: String,
    pub sort_order: i32,
}
